### ISSUE TITLE
Add-AzLoadBalancerInboundNatRuleConfig not working

### ISSUE THREAD

**Issue by iricigor** [Azure/azure-powershell#8794]
## Description

Commandlet `Add-AzLoadBalancerInboundNatRuleConfig` and others related to NAT rules are strange. They throw an error about missing non-existing parameter (probably something internal).

Also, you cannot specify backend IP configuration in any of them, but it is required in reality. How the load balancer will know where to send traffic that is matching this rule?

This is valid also in `AzureRM.Network` version `6.11.1`. Though this seems to be long running problem, the issue was also reported long time back, for example 2 years back in #2910.

## Steps to reproduce

```powershell
$LB = Get-AzLoadBalancer -ResourceGroupName 'MyRG'
$LB | Add-AzLoadBalancerInboundNatRuleConfig -Name 10320 -BackendPort 22 -FrontendPort 10320 -Protocol Tcp -FrontendIpConfigurationId ($LB.FrontendIpConfigurations[0].Id) | Set-AzLoadBalancer
```

## Environment data

```
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.1.3
PSEdition                      Core
GitCommitId                    6.1.3
OS                             Linux 4.4.0-18356-Microsoft #1-Microsoft Fri Mar 08 16:07:00 PST 2019
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

## Module versions

```powershell
Get-Module Az.Network -List | Select Name, Version

Name       Version
----       -------
Az.Network 1.3.0
```

## Debug output

```
DEBUG: 1:22:40 PM - AddAzureRmLoadBalancerInboundNatRuleConfigCommand begin processing with ParameterSet 'SetByResourceId'.
DEBUG: 1:22:40 PM - using account id '***'...
DEBUG: 1:22:40 PM - SetAzureLoadBalancerCommand begin processing with ParameterSet '__AllParameterSets'.
DEBUG: 1:22:40 PM - using account id '***'...
Add-AzLoadBalancerInboundNatRuleConfig : Value cannot be null.
Parameter name: source
At line:1 char:7
+ $LB | Add-AzLoadBalancerInboundNatRuleConfig -Name 10320 -BackendPort ...
+       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : CloseError: (:) [Add-AzLoadBalancerInboundNatRuleConfig], ArgumentNullException
+ FullyQualifiedErrorId : Microsoft.Azure.Commands.Network.AddAzureRmLoadBalancerInboundNatRuleConfigCommand

DEBUG: AzureQoSEvent: CommandName - Add-AzLoadBalancerInboundNatRuleConfig; IsSuccess - False; Duration - 00:00:00.0720651; Exception - System.ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.Linq.Enumerable.SingleOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
   at Microsoft.Azure.Commands.Network.AddAzureRmLoadBalancerInboundNatRuleConfigCommand.Execute()
   at Microsoft.Azure.Commands.Network.NetworkBaseCmdlet.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord();
DEBUG: Finish sending metric.
DEBUG: 1:22:41 PM - AddAzureRmLoadBalancerInboundNatRuleConfigCommand end processing.
DEBUG: 1:22:41 PM - AddAzureRmLoadBalancerInboundNatRuleConfigCommand end processing.
DEBUG: AzureQoSEvent: CommandName - Set-AzLoadBalancer; IsSuccess - True; Duration - 00:00:00.7253927; Exception - ;
DEBUG: Finish sending metric.
DEBUG: 1:22:41 PM - SetAzureLoadBalancerCommand end processing.
DEBUG: 1:22:41 PM - SetAzureLoadBalancerCommand end processing.
```

## Error output

Please note that there is no parameter named `source` in the commandlet.

```
Add-AzLoadBalancerInboundNatRuleConfig : Value cannot be null.
Parameter name: source
At line:1 char:7
+       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : CloseError: (:) [Add-AzLoadBalancerInboundNatRuleConfig], ArgumentNullException
+ FullyQualifiedErrorId : Microsoft.Azure.Commands.Network.AddAzureRmLoadBalancerInboundNatRuleConfigCommand
```

```
Message        : Value cannot be null.
                 Parameter name: source
StackTrace     :    at System.Linq.Enumerable.SingleOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
                    at Microsoft.Azure.Commands.Network.AddAzureRmLoadBalancerInboundNatRuleConfigCommand.Execute()
                    at Microsoft.Azure.Commands.Network.NetworkBaseCmdlet.ExecuteCmdlet()
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
Exception      : System.ArgumentNullException
InvocationInfo : {Add-AzLoadBalancerInboundNatRuleConfig}
Line           : $LB | Add-AzLoadBalancerInboundNatRuleConfig -Name 10320 -BackendPort 22 -FrontendPort 10320 -Protocol Tcp -FrontendIpConfigurationId ($LB.FrontendIpConfigurations[0].Id) | Set-AzLoadBalancer
Position       : At line:1 char:7
                 + $LB | Add-AzLoadBalancerInboundNatRuleConfig -Name 10320 -BackendPort ...
                 +       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 70
```

## Portal screenshot

From the portal is also visible that backend IP configuration is required.

![image](https://user-images.githubusercontent.com/15923171/54525908-7f76bc80-4975-11e9-94b7-9da2f9874994.png)

**Comment by iricigor** [comment]
And if adding inbound NAT rule configuration requires some other commands also, it might be good to mention it in the [help page for the commandlet](https://docs.microsoft.com/en-us/powershell/module/az.network/add-azloadbalancerinboundnatruleconfig).

**Comment by vladimir-shcherbakov** [comment]
@gmainar Could you please take a look?

**Comment by jesseconnr** [comment]
This is still a problem and I have yet to find a way to make it work. The result is you have to go back to the UI and manually configure each rule with the correct IP, which takes ~30-45 seconds for each update and you can only modify them one at a time. If you have a lot of rules, this is a nightmare.

**Comment by irenehua** [comment]
Hi,

I tried to reproduce the issue but there was no issue for me. Below are the steps taken.

1. I created Standard Public LB with a Public IP for frontend IP configuration on Azure portal and added a VM to the backend pool. 
2. I ran the 2 commands provided by @iricigor 
![image](https://user-images.githubusercontent.com/59070966/73317253-4a203980-41ea-11ea-9540-0924fb15e0ec.png) and waited about 5-8 seconds.
3. Inbound NAT rule shows on Azure portal under this LB.
![image](https://user-images.githubusercontent.com/59070966/73317307-7c319b80-41ea-11ea-8ee4-fe885fb62311.png)

Let me know the differentiation between our approaches and I can re-investigate. 

-Irene
